sympy/solvers/solvers.py

```diff
@@ -361,6 +361,50 @@ def checksol(f, symbol, sol=None, **flags):
     # TODO: improve solution testing
 
 
+def failing_assumptions(expr, against=None, **assumptions):
+    """
+    Helper for ``check_assumptions``.
+
+    Returns empty list `[]` if all assumptions are True, or a list
+    of  assumption(s) that the expression does not satisfy or
+    `None` if it can't conclude.
+
+    Examples
+    ========
+
+        >>> from sympy import Symbol
+        >>> from sympy.solvers.solvers import failing_assumptions
+        >>> failing_assumptions(-5, integer=False, positive=True)
+        ['integer', 'positive']
+
+        >>> x = Symbol('x', positive=True)
+        >>> y = Symbol('y')
+        >>> failing_assumptions(6*x, positive=True)
+        []
+
+        >>> failing_assumptions(y, real=True)
+    """
+    if against is not None:
+        assumptions = against.assumptions0
+
+    expr = sympify(expr)
+
+    flag = True
```

Comment:
`flag` is used to see whether there are any failing assumptions or can `expr` be concluded. (If it can't conclude make it `False`, but after if it finds some assumptions to be failing then make it `True`).